test/table.js

```diff
@@ -1378,6 +1378,248 @@ describe('Bigtable/Table', function() {
+        clock.restore();
+      });
+
+      it('should return true if token is consistent', done => {
+        responses = [
+          (config, callback) =>
+            callback(null, {consistency_token: 'sample-token12345'}),
+          (config, callback) => callback(null, {consistent: true}),
+        ];
+
+        table.waitForReplication(function(err, response) {
+          // Checks that a 10 minute timer was set.
+          setTimeoutSpy.calledWith(sinon.match.func, 10 * 60 * 1000);
+
+          // check checkConsistencySpy called for first time
+          assert.strictEqual(checkConsistencySpy.callCount, 1);
+
+          // Checks that clearInterval was called.
+          assert.strictEqual(clearTimeoutSpy.callCount, 1);
+
+          assert.strictEqual(response, true);
+          assert.ifError(err);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should retry checkConsistency', done => {
+        responses = [
+          (config, callback) =>
+            callback(null, {consistency_token: 'sample-token12345'}),
+          (config, callback) => callback(null, {consistent: false}),
+          (config, callback) => callback(null, {consistent: true}),
+        ];
+
+        table.waitForReplication(function(err, response) {
+          // Checks that a 10 minute timer was set.
+          setTimeoutSpy.calledWith(sinon.match.func, 10 * 60 * 1000);
+
+          // check checkConsistencySpy called for first time
+          checkConsistencySpy.callOnce;
+
+          setTimeoutSpy.calledWith(sinon.match.func, 5000);
+
+          // check checkConsistencySpy called twice after 5seconds
+          clock.tick(5010);
+          assert.strictEqual(checkConsistencySpy.callCount, 2);
+
+          // Checks that clearInterval was called.
+          setTimeoutSpy.called;
+          assert.ifError(err);
+          assert.strictEqual(response, true);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should return false after 10 min if inconsistency repeats', done => {
+        this.timeout(10 * 61 * 1000); // 10 minutes
+
+        table.bigtable.request = function(config, callback) {
+          if (config.method === 'generateConsistencyToken') {
+            return callback(null, {consistency_token: 'sample-token12345'});
+          }
+          if (config.method === 'checkConsistency') {
+            return callback(null, {consistent: false});
+          }
+        };
+
+        table.waitForReplication(function(err, response) {
+          setTimeoutSpy.called;
+          clock.tick(10 * 60 * 1000); // 10 minutes
+          assert.strictEqual(response, false);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should return error if checkonsistency returns error', done => {
+        this.timeout(10 * 61 * 1000); // 10 minutes
+
+        let error = new Error('consistency-check error');
+
+        table.bigtable.request = function(config, callback) {
```

Comment:
Please use the responses pattern as above since it's self documenting which method should be called first. If you'd like you can add an assert for the method name in the callback

Reply:
Done https://github.com/googleapis/nodejs-bigtable/pull/82/commits/5ba4e36e0e7b47a136f301cf7d9a01473e68add6